src/AllowList.sol

```diff
@@ -0,0 +1,140 @@
+     */
+    modifier limitBytesSize(bytes memory data) {
+        require(data.length <= MAX_BYTES_SIZE, ByteSizeExceeded(data.length));
+        _;
+    }
+
+    /**
+     * @notice Registers a TEE workload with a specific Ethereum address in the AllowList
+     * @notice The TEE must be registered with a quote whose validity is verified by the attestationContract
+     * @dev In order to mitigate DoS attacks, the quote must be less than 20KB
+     * @param rawQuote The raw quote from the TEE device. Must be a V4 TDX quote
+     */
+    function registerTEEService(bytes calldata rawQuote) external limitBytesSize(rawQuote) {
+        (bool success, bytes memory output) = attestationContract.verifyAndAttestOnChain(rawQuote);
+
+        if (!success) {
+            revert InvalidQuote(output);
+        }
+        // at this point we know this quote was generated by an update-to-date TEE, but we still
+        // need to make sure we can parse the workloadId and ethereum public key from the quote
+
+        // now we can safely decode the output into the TDX report body, from which we can extract
+        // the ethereum public key and compute the workloadID
+        TD10ReportBody memory td10ReportBodyStruct = QuoteParser.parseTD10ReportBody(output);
+
+        // extract the ethereum public key from the quote
+        address ethAddress = QuoteParser.extractEthereumAddress(td10ReportBodyStruct);
+
+        // extract the workloadId from the quote
+        WorkloadId workloadId = QuoteParser.extractWorkloadId(td10ReportBodyStruct);
+
+        // Register the address in the allowlist with the raw quote for future quote re-verification
+        bool previouslyRegistered = addAddress(workloadId, ethAddress, rawQuote);
+
+        emit TEEServiceRegistered(ethAddress, workloadId, rawQuote, previouslyRegistered);
+    }
+
+    /**
+     * @notice Adds a TEE to the allowlist
+     * @dev It's possible that a TEE has already registered with this ethereum address, but with a different workloadId.
+     * This is expected if the TEE gets restarted or upgraded and generates a new workloadId.
+     * It's also possible that the ethereum address and workloadId are the same, but the quote
+     * is different. This is expected if Intel releases a new set of DCAP Endorsements (i.e.
+     * a new TCB), in which case the quotes the TEE generates will be different.
+     * In both cases, we need to update the allowlist with the new quote.
+     * @param workloadId The workloadId of the TEE
+     * @param ethAddress The Ethereum address of the TEE
+     * @param rawQuote The raw quote from the TEE device
+     * @return previouslyRegistered Whether the TEE was previously registered
+     */
+    function addAddress(WorkloadId workloadId, address ethAddress, bytes calldata rawQuote)
+        internal
+        returns (bool previouslyRegistered)
+    {
+        // check if the TEE is already registered with the same ethereum address and workloadId,
```

Comment:
this comment has an error, should be same workloadId and quote, at least that's whats being compared down below.

I guess we would need to compare address and quote, as workloadid is just a subset of the quote. Yet, what do we gain? the expensive part already happened (verifying the quote). Not sure if we should do this check at all?